Index: conf/battle/misc.conf
===================================================================
--- conf/battle/misc.conf	(revision 671)
+++ conf/battle/misc.conf	(working copy)
@@ -135,3 +135,10 @@
 // Good for giving special deal events to players. Prices will not go below 1 point.
 // Only works with a set PACKETVER and client date of 2010-08-03 or higher.
 cashshop_price_rate: 100
+
+// Whether or not mail box status is displayed upon login.
+// Default: 0
+// 0 = No
+// 1 = Yes
+// 2 = Yes, when there are unread mails
+mail_show_status: 0
Index: src/map/intif.c
===================================================================
--- src/map/intif.c	(revision 671)
+++ src/map/intif.c	(working copy)
@@ -1466,7 +1466,7 @@
 
 	if (flag)
 		clif_Mail_refreshinbox(sd);
-	else
+	else if( battle_config.mail_show_status && ( battle_config.mail_show_status == 1 || sd->mail.inbox.unread ) )
 	{
 		char output[128];
 		sprintf(output, msg_txt(510), sd->mail.inbox.unchecked, sd->mail.inbox.unread + sd->mail.inbox.unchecked);
