Index: security.c
===================================================================
--- security.c	(revision 187)
+++ security.c	(working copy)
@@ -55,6 +55,8 @@
 	S_CANT_SEND_GINVITE = 0x0400,		// Cannot Send Guild Invite
 	S_CANT_RECEIVE_GINVITE = 0x0800,	// Cannot Receive Guild Invite
 	S_CANT_LEAVE_GUILD = 0x1000,		// Cannot Leave Guild
+	S_CANT_USEITEM = 0x2000,			// Cannot Use Item
+	S_CANT_CLICK = 0x4000,				// Cannot Click Npc
 };
 
 int pc_cant_drop(struct map_session_data **sd, int *n, int *amount) {	// Can't Drop Items
@@ -68,6 +70,18 @@
 	return 1;	
 }
 
+int pc_cant_useitem(struct map_session_data **sd,int *n) {
+	if (*sd == NULL)
+		return 1;
+	if (is_secure(*sd)){
+		if (security_opt(*sd)&S_CANT_USEITEM){
+			clif->message((*sd)->fd,"Security is on. You cannot Use item.");
+			hookStop();
+			return 1;
+		}
+	}
+	return 0;
+}
+
 void cant_trade(struct map_session_data **sd, struct map_session_data **target_sd){	// Can't receive/send Trade Requests
 	if (*sd == NULL || *target_sd == NULL)
 		return;
@@ -167,6 +181,22 @@
 	return 0;
 }
 
+int npc_cant_click(struct map_session_data **sd, struct npc_data **nd) {
+	if (*sd == NULL)
+		return 1;
+	if (is_secure(*sd)){
+		if (security_opt(*sd)&S_CANT_CLICK){
+			clif->message((*sd)->fd,"Security is on. You cannot Click Npc.");
+			hookStop();
+			return 1;
+		}
+	}
+	return 0;
+}
+
 void open_vending(struct map_session_data **sd, int *num){
 	if (*sd == NULL)
 		return;
@@ -219,6 +249,7 @@
 HPExport void plugin_init (void)
 {
 	addHookPre(pc, dropitem, pc_cant_drop);
+	addHookPre(pc, useitem, pc_cant_useitem);
 	addHookPre(trade, request, cant_trade);
 	addHookPre(gstorage, open, gstorage_cant_open);
 	addHookPre(gstorage, additem, gstorage_cant_add);
@@ -230,6 +261,7 @@
 	addHookPre(clif, openvendingreq, open_vending);
 	addHookPre(guild, invite, guild_invite_permission);
 	addHookPre(guild, leave, guild_leave_permission);
+	addHookPre(npc, click, npc_cant_click);
 }
 
 HPExport void server_online (void) {
